lib/sgtn-client/api/base.rb

```diff
@@ -0,0 +1,92 @@
+# 
+#  Copyright 2019-2022 VMware, Inc.
+#  SPDX-License-Identifier: EPL-2.0
+#
+
+require 'multi_json'
+
+module SgtnClient
+
+  class Base
+    class << self
+
+      def getTranslation(component, key, locale)
+        locale = SgtnClient::LocaleUtil.get_best_locale(locale)
+        items = get_cs(component, locale)
+        if items.nil? || items["messages"].nil?
+          nil
+        else
+          items["messages"][key]
+        end
+      end
+
+      def compareSource(component, key, default_locale, source, translation)
+        SgtnClient.logger.debug "[Base][compareSource]component=#{component},key=#{key},default_locale=#{default_locale},source=#{source},translation=#{translation}"
+        items = get_cs(component, default_locale)
+        if items.nil? || items["messages"].nil?
+          translation
+        else
+          target = items["messages"][key]
+          source == target ? translation : source
+        end
+      end
+
+      def get_cs(component, locale)
+        cache_key = SgtnClient::CacheUtil.get_cachekey(component, locale)
+        SgtnClient.logger.debug "[Base][get_cs]cache_key=#{cache_key}"
```

Comment:
Log what it's doing here rather than only a cache key.

Reply:
the method name [get_cs] tell what actually